internal/controller/dws_storage_controller.go

```diff
@@ -190,22 +202,26 @@ func (r *DWSStorageReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			storage.Status.Message = ""
 		}
 
-		if node.Status.Fenced {
+		if nnfNode.Status.Fenced {
 			storage.Status.Status = dwsv1alpha2.DegradedStatus
 			storage.Status.RebootRequired = true
 			storage.Status.Message = "Storage node requires reboot to recover from STONITH event"
 		} else {
-			ready, err := r.isKubernetesNodeReady(ctx, storage)
+			nodeState, err := r.coreNodeState(ctx, storage)
 			if err != nil {
 				return ctrl.Result{}, err
 			}
 
-			if !ready {
+			if !nodeState.nodeReady {
 				log.Info("storage node is offline")
 				storage.Status.Status = dwsv1alpha2.OfflineStatus
 				storage.Status.Message = "Kubernetes node is offline"
+			} else if len(nodeState.nnfTaint) > 0 {
+				log.Info(fmt.Sprintf("storage node is tainted with %s", nodeState.nnfTaint))
+				storage.Status.Status = dwsv1alpha2.DisabledStatus
```

Comment:
Should this be something other than "disabled" like "drained"? Disabled is used when the Spec.State field is set to Disabled. What is adding the taint supposed to signify that setting the node to disabled doesn't do?

Reply:
Sure, I could do "drained".

The taint is documented here: https://nearnodeflash.github.io/latest/guides/node-management/drain/

It removes the nnf pods from the node, while leaving the lustre-csi-driver in place to do the unmount of any filesystems. Prior to this, if we just tainted the node to chase everything off, sometimes k8s would remove the lustre-csi-driver first, and then the nnf pods would be stuck because k8s couldn't get an answer when it asked for the filesystems to be unmounted.